src/ssl.c

```diff
@@ -18845,6 +18868,13 @@ WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_set_peer_cert_chain(WOLFSSL* ssl)
     if (sk == NULL) {
         WOLFSSL_MSG("Null session chain");
     }
+#if defined(WOLFSSL_QT)
+    else if (ssl->options.side == WOLFSSL_SERVER_END) {
```

Comment:
Is this Qt-specific, or just OpenSSL specific?  I'm wondering if this should be protected by a define more generic than WOLFSSL_QT.  OPENSSL_EXTRA/OPENSSL_ALL?

Reply:
This is OpenSSL specific, and this is for server side only. When we look at tls_process_server_certificate(), the comments say
"Inconsistency alert: cert_chain does include the peer's certificate, which we don't include in statem_srvr.c". This means client  side chain includes peer's certificate, but not server side.

Changed guard from WOLFSSL_QT to OPENSSL_ALL